src/dfcx_scrapi/builders/intent.py

```diff
@@ -14,19 +14,467 @@
+        elif mode in ["phrases", "training phrases"]:
+            print(self._show_training_phrases(repeat_count=repeat_count))
+        elif mode == "whole":
+            print(self.__str__())
+        else:
+            raise ValueError(
+                "mode should be in"
+                " ['basic', 'parameters',"
+                " 'phrases', 'training phrases', 'whole']"
+            )
+
+
+    def show_stats(self):
+        """Provide some stats about the intent."""
+        repeat_count_dict = defaultdict(lambda: 0)
+        annotated_count = 0
+        for tp in self.proto_obj.training_phrases:
+            repeat_count_dict[tp.repeat_count] += 1
+
+            for part in tp.parts:
+                if part.parameter_id:
+                    annotated_count += 1
+                    break
+
+        phrases_count = len(self.proto_obj.training_phrases)
+        params_count = len(self.proto_obj.parameters)
+        annotated_pct = round(100*annotated_count/phrases_count, 1)
+        uniques_count = repeat_count_dict[1]
+        uniques_pct = round(100*uniques_count/phrases_count, 1)
+        non_uniques_count = phrases_count - uniques_count
+        non_uniques_pct = round(100*non_uniques_count/phrases_count, 1)
+
+        phrases_str = f"# of training phrases: {phrases_count}"
+        params_str = f"# of parameters: {params_count}"
+        annotated_str = (
+            "Annotated training phrases:"
+            f" {annotated_pct}% ({annotated_count})"
+        )
+
+        uniques_str = (
+            "Unique training phrases:"
+            f" {uniques_pct}% ({uniques_count})"
+        )
+        non_uniques_str = (
+            "Non-unique training phrases:"
+            f" {non_uniques_pct}% ({non_uniques_count})"
+        )
+
+        repeat_count_srt = "\n\t".join([
+            f"with repeat count {i}: {repeat_count_dict[i]}"
+            for i in sorted(repeat_count_dict.keys())
+        ])
+
+        out = (
+            f"{phrases_str}\n{annotated_str}\n{params_str}\n"
+            f"\n{uniques_str}\n{non_uniques_str}\n\t{repeat_count_srt}"
+        )
+        print(out)
+
+
+    def parameter_checking(self) -> bool:
+        """Check if the annotated parameters exist
+        in the Parameter attribute of proto_obj.
+        """
+        tp_params_set = set()
+        for tp in self.proto_obj.training_phrases:
+            for part in tp.parts:
+                tp_params_set.add(part.parameter_id)
+        # Remove the empty string for unannotated parts
+        try:
+            tp_params_set.remove("")
+        except KeyError:
+            pass
+
+        # Get the parameters from proto_obj
+        parameters_set = {param.id for param in self.proto_obj.parameters}
+
+        # Check for not existing annotated parameters
+        for tp_param in tp_params_set:
+            if tp_param not in parameters_set:
+                raise Exception(
+                    f"parameter_id {tp_param} does not exist in parameters."
```

Comment:
@MRyderOC wrap `{tp_param}` in back ticks so it stands out in the error message. 

For example, imagine I created a parameter called `id`. The error message would look like this:
```
parameter_id id does not exist in parameters.
```

Instead, if we add back ticks it can be a bit more pronounced like:
```
parameter_id `id` does not exist in parameters.
```